xerxes.md

```diff
@@ -234,6 +236,28 @@ which lists each statistic, the slots a, b, c and d, the number of sites with no
 
 Additionally, an option `--blockOutFile` can be specified, to which then a table with estimates per Jackknife block is written.
 
+#### Degenerate statistics
+
+Specific cases of statistics are 0 by construction:
+
+- `F2(A, B)`, `F2vanilla(A, B)`, `FST(A, B)` and `FSTvanilla(A, B)` where `A=B`.
+- `F3(A, B, C)` and `F3vanilla(A, B, C)` where `C=A` or `C=B`
+- `F4(A, B, C, D)` where `A=B` or `C=D`
+
+Even though the bias-correction technically can result in non-zero and even negative values, we automatically detect these cases and output identical 0 for them. This can be useful for example when looping over pairs of populations for a pairwise matrix of FST, where we then want the diagonal to be zero to yield a proper distance matrix.
+
+#### Ploidy and illegal cases
+
+Genotype ploidy in input samples is important for many of the statistics, because the bias-correction terms require the number of chromosomes. Ploidy information is automatically read through the field of `Genotype_Ploidy` in the .janno file. A warning is printed if that information is missing, in which case we assume diploid genotypes. But often with low-coverage data from ancient DNA we create pseudo-haploid genotypes, so in that case it is important to provide that information correctly through the .janno file. 
+
+In specific cases, statistics are illegal, in case of only a single haplotype. Specifically:
+
+- `F2(A, B)` and `FST(A, B)` is undefined if either one of `A` or `B` contains only a single haplotype.
+- `F3(A, B, C)` is undefined of `C` contains only a single haplotype.
+- `Het(A)` unsurprisingly is undefined if `A` contains only a single haplotype.
+
+These cases are detected and an error is thrown. For of `F2`, `F3` and `FST` it suggests to use the "vanilla" versions of the statistics if that makes sense. This is particularly relevant for so-called "Outgroup-F3-Statistics", where we sometimes use a single haploid reference genome in position `C`. Use `F3vanilla` in that case.
+
 #### Whitepaper
 The repository comes with a [detailed whitepaper](https://github.com/poseidon-framework/poseidon-analysis-hs/blob/updates_poseidon_1.4/docs/xerxes_whitepaper.pdf) that describes some more mathematica details of the methods implemented here.
```

Comment:
This should probably point to the version of the whitepaper on the master/main branch.

Reply:
Yes, of course. Fixed.